### PR TITLE
docs(geo): add documentation for JS geofences developer preview

### PR DESCRIPTION
**Pull request by TreTuna**
_Description of changes:_
Adds the documentation for the developer preview of `@aws-amplify/geo` Geofences features.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by offlineprogrammer — APPROVED** [review]

File: src/directory/directory.js [hunk 1]
```diff
@@ -448,6 +448,11 @@ const directory = {
             route: "/lib/geo/search",
             filters: ["js"],
           },
+          {
+            title: "Geofences (Developer preview)",
```
Comment (Amplifiyer) [inline src/directory/directory.js:452]: nit: CLI says it Geofencing

File: src/directory/directory.js [hunk 2]
```diff
@@ -1223,6 +1228,11 @@ const directory = {
             route: "/cli/geo/search",
             filters: [],
           },
+          {
+            title: "Geofencing (Developer Preview)",
```
Comment (Amplifiyer) [inline src/directory/directory.js:1232]: nit: Library says it Geofences

File: src/fragments/lib/geo/js/geofences.mdx [hunk 3]
```diff
@@ -0,0 +1,315 @@
+<Callout>
+
+**Note:** Please reach out to us for any feedback and/or issues [here](https://github.com/aws-amplify/amplify-js/issues)
+
+</Callout>
+
+## Provisioning geofence resources
+First, make sure you've provisioned a geofence collection resource and configured your app using the instructions in either [Amplify CLI - Geo - Geofence Collection](/cli/geo/geofences) or [Use existing Amazon Location Service resources](/lib/geo/existing-resources) and you have already setup [displaying a map](/lib/geo/maps) in your application.
```
Comment (Amplifiyer) [inline src/fragments/lib/geo/js/geofences.mdx:8]: This link looks wrong, shouldn't it be `/cli/geo/geofencing`

File: src/fragments/lib/geo/js/geofences.mdx [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+First, create a map onto which you want to add the geofence management component. See the guide on [creating and displaying maps](https://docs.amplify.aws/lib/geo/maps/q/platform/js/).
+
+Then, import [AmplifyGeofenceControl](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/geo/API.md#amplifygeofencecontrol) from "maplibre-gl-js-amplify", create a new instance of this control and add it to your MapLibre map instance.
+
+Your updated map application should look like the following
+
+> **Note:** Ensure that your package bundler (webpack, rollup, etc) is configured to handle css files. Check out the webpack documentation [here](https://webpack.js.org/loaders/css-loader/).
+
+> **Notes:** To use Geofence Controls the user will need to be authenticated with the administrative Cognito user associated with the Geofence Collection you created above. Below is an example using React and the [Amplify Authenticator](https://ui.docs.amplify.aws/components/authenticator) but for other use cases check the [Authentication documentation](/lib/auth/overview).
+
```
Comment (Amplifiyer) [inline src/fragments/lib/geo/js/geofences.mdx:31]:
```suggestion

> **Note:** Ensure that your package bundler (webpack, rollup, etc) is configured to handle css files. Check out the webpack documentation [here](https://webpack.js.org/loaders/css-loader/).

> **Notes:** To use Geofence Controls the user will need to be authenticated with the administrative Cognito user associated with the Geofence Collection you created above. Below is an example using React and the [Amplify Authenticator](https://ui.docs.amplify.aws/components/authenticator) but for other use cases check the [Authentication documentation](/lib/auth/overview).

Your updated map application should look like the following

```

File: src/fragments/lib/geo/js/geofences.mdx [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
++ }
+```
+
+![Geofence Create](/images/geofence-create.png)
+
+![Geofence Edit](/images/geofence-edit.png)
+
+## Geofence API
+
+If you are using a different mapping library or have a need for a programmatic approach to managing geofences, the `@aws-amplify/geo` package provides methods for managing geofences, but not geofence collections.
```
Comment (Amplifiyer) [inline src/fragments/lib/geo/js/geofences.mdx:106]:
```suggestion
If you are using a different mapping library or need a programmatic approach to managing geofences, the `@aws-amplify/geo` package provides methods for managing geofences, but not geofence collections.
```

File: src/fragments/lib/geo/js/geofences.mdx [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+```diff
+import { useEffect, useRef } from "react";
+- import { createMap } from "maplibre-gl-js-amplify";
++ import { createMap, AmplifyGeofenceControl } from "maplibre-gl-js-amplify";
++ import { withAuthenticator } from "@aws-amplify/ui-react";
++ import "@aws-amplify/ui-react/styles.css";
+import "maplibre-gl/dist/maplibre-gl.css";
+
+function Map() {
```
Comment (thaddmt) [inline src/fragments/lib/geo/js/geofences.mdx:40]:
```suggestion
function Map({ signOut }) {
```

File: src/fragments/lib/geo/js/geofences.mdx (hunk 3 above)
Comment (thaddmt) [inline src/fragments/lib/geo/js/geofences.mdx:8]:
```suggestion
First, make sure you've provisioned a geofence collection resource and configured your app using the instructions in either [Amplify CLI - Geo - Geofence Collection](/cli/geo/geofencing) or [Use existing Amazon Location Service resources](/lib/geo/existing-resources) and you have already setup [displaying a map](/lib/geo/maps) in your application.
```

**Review by thaddmt — APPROVED** [review]

**Pull request merged**